packages/data-context/src/sources/MigrationDataSource.ts

```diff
@@ -221,48 +244,39 @@ export class MigrationDataSource {
   }
 
   private async initializeFlags () {
+    if (!this.ctx.currentProject) {
+      throw Error('Need currentProject to do migration')
+    }
+
     const config = await this.parseCypressConfig()
 
-    this.hasCustomIntegrationSpecPattern = config.testFiles !== undefined || config.e2e?.testFiles !== undefined
+    this.hasCustomIntegrationTestFiles = getIntegrationTestFiles(config) !== '**/*'
+    this.hasCustomIntegrationFolder = getIntegrationFolder(config) !== 'cypress/integration'
 
-    this.hasCustomIntegrationFolder = config.e2e?.integrationFolder !== undefined || config.integrationFolder !== undefined
+    const componentFolder = getComponentFolder(config)
 
-    this.hasCustomComponentSpecPattern = config.component?.testFiles !== undefined || config.testFiles !== undefined
+    this.hasCustomComponentFolder = componentFolder !== 'cypress/component'
 
-    this.hasCustomComponentFolder = config.component?.componentFolder !== undefined || config.componentFolder !== undefined
+    const componentTestFiles = getComponentTestFiles(config)
 
-    // TODO: implement this properly
-    this.hasComponentTesting = true
-  }
+    this.hasCustomComponentTestFiles = componentTestFiles !== '**/*'
 
-  private shouldShowStep (step: MIGRATION_STEP): boolean {
-    switch (step) {
-      case 'renameAuto': return !(this.hasCustomIntegrationSpecPattern && this.hasCustomComponentSpecPattern)
-      case 'renameManual': return this.hasComponentTesting
-      case 'setupComponent': return this.hasComponentTesting
-      default: return true
+    if (componentFolder === false) {
+      this.hasComponentTesting = false
```

Comment:
This isn't necessarily true is it? They could have component configuration?

Reply:
componentFolder is defined as `config.component.componentFolder ?? config.component ?? 'cypress/component'`. Which makes me think this if statement will never be reached

Reply:
You could be right. Although I'm not sure why you'd have CT configured and then turn off your componentFolder - you wouldn't be able to load any specs, but it's an edge case we should consider. I wonder what the best way to def decide they are using component testing is? Bart has this idea https://github.com/cypress-io/cypress/pull/19909/files#diff-b7132f5eaea57fbe5cde51330fafe8916684f9add91fa3a6b48ac61e47a1634eR230 (check for any component specs in the project). Thoughts?

Reply:
I think this assumption *is* reasonable, after thinking about it more. If `componentFolder` is `false`, we cannot look to see if they have any component specs (how to we know where to look? The user is intentionally saying "I do not have any component specs", thus there is nothing to migrate. We will still add `componentFolder: false` to their generated `cypress.config.js`, and we can still migrate their `component.supportFile`, if they have one.

Reply:
I made the checks more explicit https://github.com/cypress-io/cypress/pull/19909/commits/2c345497e08e3641e1fbbbcfcc8a75d68c4f6ffe